bin/generate_gene_counts.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+from collections import OrderedDict
+from pathlib import Path
+
+import pandas as pd
+
+translator = {
+    "unstranded": 1,
+    "forward": 2,
+    "reverse": 3,
+}
+
+
+def get_non_std_genes(gtf: str) -> set[str]:
+    """Create list of genes not belonging to chr1-21 or chrM"""
+
+    std_chrs = [
+        "chr1",
+        "chr2",
+        "chr3",
+        "chr4",
+        "chr5",
+        "chr6",
+        "chr7",
+        "chr8",
+        "chr9",
+        "chr10",
+        "chr11",
+        "chr12",
+        "chr13",
+        "chr14",
+        "chr15",
+        "chr16",
+        "chr17",
+        "chr18",
+        "chr19",
+        "chr20",
+        "chr21",
+        "chr22",
+        "chrX",
+        "chrY",
+        "chrM",
+    ]
+
+    gene_id_regex = re.compile('gene_id "(.+?)"')
+    genes_to_exclude = []
+
+    with open(gtf, "r") as gtf_file:
+        for line in gtf_file:
+            if line.startswith("#"):
+                continue
+
+            if line.split()[0] in std_chrs:
+                continue
+
+            gene_id = re.search(gene_id_regex, line)
+            genes_to_exclude.append(gene_id.group(1))
+
+    return set(genes_to_exclude)
+
+
+def read_star_gene_cnts(sample: str, star: Path, strandedness: str) -> dict:
+    """Read gene count file(s) from STAR output."""
+    sample_ids = {}
+    gene_ids = {}
+    with open(star) as in_tab:
+        for line in in_tab:
+            if not line.startswith("N_"):
+                gene_id = line.split()[0]
+                strand = translator[strandedness.lower()]
+                counts = line.split()[strand]
+                gene_ids[gene_id] = int(counts)
+    gene_ids = OrderedDict(sorted(gene_ids.items()))
+    sample_ids[sample] = gene_ids
+    return sample_ids
+
+
+def transform_to_table(gene_ids_dict, outfile, genes_to_exlude, ref_count_file: Path):
```

Comment:
add type hints

Reply:
Can't see that `genes_to_exclude` is used here

Reply:
Will do! The genes_to_exclude is used in line 109

Reply:
ah now I see it 🙈